tests/helpers/FakeEthLightwallet.ts

```diff
@@ -0,0 +1,293 @@
+        keystore,
+        password,
+        callback
+      ) => this._upgradeOldSerialized(
+        keystore,
+        password,
+        callback
+      )
+    }
+  }
+
+  /**
+   * Sets flag to mock an error for specific method.
+   * @param methodName Name of method to set error flag for.
+   */
+  public setError (methodName) {
+    this.errors[methodName] = true
+  }
+
+  /**
+   * Removes flags to mock errors for all methods.
+   */
+  public removeErrors () {
+    this.errors = []
+  }
+
+  /**
+   * Mocked lightwallet.txutils methods
+   */
+  private _functionTx (
+    abi,
+    functionName,
+    args,
+    options
+  ) {
+    if (this.errors['functionTx']) {
+      throw new Error('Mocked error in lightwallet.txutils.functionTx()')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+  private _valueTx (options) {
+    if (this.errors['valueTx']) {
+      throw new Error('Mocked error in lightwallet.txutils.valueTx()')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+
+  /**
+   * Mocked lightwallet.signing methods
+   */
+  private _signTx (
+    keystore,
+    pwDerivedKey,
+    rlpHexTx,
+    address
+  ) {
+    if (this.errors['signTx']) {
+      throw new Error('Mocked error in lightwallet.signing.signTx')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+  private _signMsgHash (
+    keystore,
+    pwDerivedKey,
+    personalMsgHashBuff,
+    address
+  ) {
+    if (this.errors['signMsgHash']) {
+      throw new Error('Mocked error in lightwallet.signing.signMsgHash')
+    }
+    return {
+      r: new Buffer('r'),
+      s: new Buffer('s'),
+      v: 24
+    }
+  }
+  private _concatSig (signature) {
+    return 'Concat signature'
+  }
+
+  /**
+   * Mocked lightwallet.keystore methods
+   */
+  private _serialize () {
+    return keystore1
+  }
+  private _deserialize (serializedKeystore) {
+    return this.keystore
+  }
+  private _keyFromPassword (password, callback) {
+    if (this.errors['keyFromPassword']) {
+      setTimeout(() => callback(new Error('Mocked error in lightwallet.keystore.keyFromPassword()'), undefined))
+      return
+    }
+    setTimeout(() => callback(undefined, this._pwDerivedKey))
+    return
+  }
+  private _exportPrivateKey (address, pwDerivedKey) {
+    if (this.errors['exportPrivateKey']) {
+      throw new Error('Mocked error in lightwallet.keystore.exportPrivateKey')
+    }
+    return this._privateKey
+  }
+  private _createVault (options, callback) {
```

Comment:
Can you do the same thing with the setTimeout here to be sure we also are save here?

Reply:
yes done